src/useConstantCallback.js

```diff
@@ -0,0 +1,30 @@
+// @flow
+import * as React from 'react'
+
+/**
+ * Creates a callback, with dependencies, that will be
+ * instance === for the lifetime of the component.
+ */
+export default function useConstantCallback(callback, deps) {
+  // initialize refs on first render
+  const refs = deps.map(React.useRef)
+  // update refs on each additional render
+  deps.forEach((dep, index) => (refs[index].current = dep))
```

Comment:
Doesn't `refs` get created at each render, this `forEach` line is not needed 🤔

Reply:
No they don't. `useRef` is similar to `useState` -- whatever is passed as it's first param only initialises the state/ref in the very first render.

Reply:
My understanding is that you shouldn't update refs in render function as it would break with Suspense. See Dan's comment on this issue: https://github.com/facebook/react/issues/21191#issuecomment-815316815

Reply:
Understood, even if the `deps.map` is called each time, it will still get a new array but with first deps

Reply:
I think he probably means why not a single ref containing the deps array, instead of a ref for each dep. Then you don't need the forEach line, or the length check.

Reply:
> I think he probably means why not a single ref containing the deps array, instead of a ref for each dep. Then you don't need the forEach line, or the length check.
I wrote this yesterday, and had this thought upon waking this morning. Why not save the whole array? 👍